drivers/dma/Kconfig.stm32

```diff
@@ -0,0 +1,33 @@
+# Kconfig - DMA configuration options
+#
+#
+# Copyright (c) 2016 Intel Corporation
+# Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig DMA_STM32
+	bool "Enable STM32 DMA driver"
+	select DYNAMIC_INTERRUPTS
+	depends on SOC_FAMILY_STM32
+	depends on HEAP_MEM_POOL_SIZE != 0
+	help
+	  DMA driver for STM32 series SoCs.
+
+if DMA_STM32
+
+config DMA_1
+	bool "Enable STM32 DMA Controller 1"
+	default y
```

Comment:
Why is it enabled by default? Is this correct?

If it is correct indeed we need then to disable it in samples/boards/96b_argonkey/microphone/prj.conf
CONFIG_DMA_1=n

Reply:
This is something I talked about with Andrzej in #13501. We thought every driver should have at least one default instance to be enabled. Please have a look in that PR, I'm not sure it's the best way for every driver, this can be discussed.

Reply:
I think it is not case for example for I2C and SPI. 
@erwango  any comment on this?
Anyway, in case you want to have a default we require to change the microphone sample
as I was saying above.

Reply:
Makes sense, example's default configuration should be in example's configuration file.

Reply:
To me, DMA instance configuration is indeed board dependent. Instance we want to activate depends on what we want to do with the board and how the resources are shared between the various tasks.
DMA_1 enabled might not be relevant for all boards. And then if some instance is enabled is makes things complicated as each board should deactivate (or not) default instance and then enable the instance they need.
So it's better to keep it default n and provide configuration on a per board basis.